helmexec/exec.go

```diff
@@ -70,60 +70,68 @@ func (helm *execer) AddRepo(name, repository, certfile, keyfile, username, passw
 }
 
 func (helm *execer) UpdateRepo() error {
 	helm.logger.Info("Updating repo")
-	out, err := helm.exec("repo", "update")
+	out, err := helm.exec([]string{"repo", "update"}, map[string]string{})
 	helm.write(out)
 	return err
 }
 
 func (helm *execer) UpdateDeps(chart string) error {
 	helm.logger.Infof("Updating dependency %v", chart)
-	out, err := helm.exec("dependency", "update", chart)
+	out, err := helm.exec([]string{"dependency", "update", chart}, map[string]string{})
 	helm.write(out)
 	return err
 }
 
 func (helm *execer) BuildDeps(chart string) error {
 	helm.logger.Infof("Building dependency %v", chart)
-	out, err := helm.exec("dependency", "build", chart)
+	out, err := helm.exec([]string{"dependency", "build", chart}, map[string]string{})
 	helm.write(out)
 	return err
 }
 
-func (helm *execer) SyncRelease(name, chart string, flags ...string) error {
+func (helm *execer) SyncRelease(context HelmContext, name, chart string, flags ...string) error {
 	helm.logger.Infof("Upgrading %v", chart)
-	out, err := helm.exec(append([]string{"upgrade", "--install", "--reset-values", name, chart}, flags...)...)
+	preArgs := context.GetTillerlessArgs(helm.helmBinary)
+	env := context.getTillerlessEnv()
+	out, err := helm.exec(append(append(preArgs, "upgrade", "--install", "--reset-values", name, chart), flags...), env)
 	helm.write(out)
 	return err
 }
 
-func (helm *execer) ReleaseStatus(name string, flags ...string) error {
+func (helm *execer) ReleaseStatus(context HelmContext, name string, flags ...string) error {
 	helm.logger.Infof("Getting status %v", name)
-	out, err := helm.exec(append([]string{"status", name}, flags...)...)
+	preArgs := context.GetTillerlessArgs(helm.helmBinary)
+	env := context.getTillerlessEnv()
+	out, err := helm.exec(append(append(preArgs, "status", name), flags...), env)
 	helm.write(out)
 	return err
 }
 
-func (helm *execer) List(filter string, flags ...string) (string, error) {
+func (helm *execer) List(context HelmContext, filter string, flags ...string) (string, error) {
 	helm.logger.Infof("Listing releases matching %v", filter)
-	out, err := helm.exec(append([]string{"list", filter}, flags...)...)
+	preArgs := context.GetTillerlessArgs(helm.helmBinary)
+	env := context.getTillerlessEnv()
+	out, err := helm.exec(append(append(preArgs, "list", filter), flags...), env)
 	helm.write(out)
 	return string(out), err
 }
 
-func (helm *execer) DecryptSecret(name string) (string, error) {
+func (helm *execer) DecryptSecret(context HelmContext, name string, flags ...string) (string, error) {
 	// Prevents https://github.com/roboll/helmfile/issues/258
 	helm.decryptionMutex.Lock()
 	defer helm.decryptionMutex.Unlock()
 
 	helm.logger.Infof("Decrypting secret %v", name)
-	out, err := helm.exec(append([]string{"secrets", "dec", name})...)
+	preArgs := context.GetTillerlessArgs(helm.helmBinary)
+	env := context.getTillerlessEnv()
+	out, err := helm.exec(append(append(preArgs, "secrets", "dec", name), flags...), env)
```

Comment:
I thought so but please let me confirm - You run helm-tiller because helm-plugin is tied to tiller, even when the actual plugin code(helm-secrets in this case, which doesnt access tiller itself) isnt?

Reply:
Exactly. Looks like helm, before calling any plugin, tries to talk to its server. Sadly inefficient.